app/facade.py

```diff
@@ -409,3 +409,29 @@ async def write_schema_definition(controller, schema_definition_request):
             detail=f"Something went wrong.\n Could not write schema to ledger.\n{schema}",
         )
     return write_schema_resp
+
+
+# Need to rename this?
+async def verify_proof_req(controller, presentation_exchange_id):
+
+    verify = await controller.proofs.verify_presentation(presentation_exchange_id)
+
+    if not verify:
+        raise HTTPException(
+            status_code=404,
```

Comment:
404 is a client error ie Not Found. Maybe a different error code would be more appropriate here. Maybe 500? although that one is pretty generic. Have a look at [this list of error codes](https://www.wikiwand.com/en/List_of_HTTP_status_codes#/4xx_client_errors)